src/test/java/org/apache/commons/lang3/SystemUtilsTest.java

```diff
@@ -373,6 +373,33 @@ public void testOSMatchesNameAndVersion() {
     }
 
     @Test
+    public void testMatchOSMajorMinorVersion() {
+        String osName = "Mac OS X";
+        String osVersion = null;
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        osVersion = "10.10";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.1"));
+        osVersion = "10.1";
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.1"));
+        osVersion = "10.10.1";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        osVersion = "10.10.10";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+    }
+
+
+    @Test
+    public void testNotConfuseMacOSXPumaAndYosemite() {
```

Comment:
Have a look at https://issues.apache.org/jira/browse/LANG-1102 and http://svn.apache.org/r1669750, I've implemented a better algorithm for matching OS version. This way it should be easy for you to implement the Mac OS X versions.